ignite/metrics/nlp/bleu.py

```diff
@@ -151,54 +168,113 @@ def _corpus_bleu(
                 p_numerators[i] += numerator
                 p_denominators[i] += denominator
 
+            # Calculate the hypothesis lengths
+            hyp_lengths += len(hyp)
+
+            # Calculate the closest reference lengths.
+            ref_lengths += _closest_ref_length(refs, len(hyp))
+
+        return (hyp_lengths, ref_lengths)
+
+    def _brevity_penalty_smoothing(
+        self, p_numerators: torch.Tensor, p_denominators: torch.Tensor, hyp_length_sum: int, ref_length_sum: int,
+    ) -> float:
+
         # Returns 0 if there's no matching n-grams
         # We only need to check for p_numerators[1] == 0, since if there's
         # no unigrams, there won't be any higher order ngrams.
         if p_numerators[1] == 0:
             return 0
 
-        # If no smoother, returns 0 if there's at least one a not matching n-grams
-        if self.smoother.smooth == "no_smooth" and min(p_numerators.values()) == 0:
+        # If no smoother, returns 0 if there's at least one a not matching n-grams]
+        if self.smoother.smooth == "no_smooth" and min(p_numerators[1:]).item() == 0:
```

Comment:
A question, shouldn't we ensure that len(p_numerators) > 1 ?

Reply:
That shouldn't be a problem as it has been initialized as `torch.zeros(self.ngrams_order + 1)`
and ngram already has a check for greater than 0 [here](https://github.com/pytorch/ignite/pull/2179/files#diff-42afc9594e3a30f4975f0edb0f1453381f0f3eb71056d5905a7f15d60eaaef3bR133-R134)